maitake/src/wait/map.rs

```diff
@@ -0,0 +1,926 @@
+use crate::{
+    loom::{
+        cell::UnsafeCell,
+        sync::{
+            atomic::{AtomicUsize, Ordering::*},
+            spin::{Mutex, MutexGuard},
+        },
+    },
+    util,
+};
+use cordyceps::{
+    list::{self, List},
+    Linked,
+};
+use core::{
+    fmt::Debug,
+    future::Future,
+    marker::PhantomPinned,
+    mem::{self, MaybeUninit},
+    pin::Pin,
+    ptr::NonNull,
+    task::{Context, Poll, Waker},
+};
+use mycelium_bitfield::FromBits;
+use mycelium_util::fmt;
+use mycelium_util::sync::CachePadded;
+use pin_project::{pin_project, pinned_drop};
+
+#[cfg(test)]
+mod tests;
+
+/// An error indicating a failed wake
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Comment:
this probably should have a 
```suggestion
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
#[non_exhaustive]
```
on it, if there's a possibility of adding new error types? not a blocker.